core/src/main/java/hudson/lifecycle/WindowsServiceLifecycle.java

```diff
@@ -120,6 +120,11 @@ public void rewriteHudsonWar(File by) throws IOException {
 
     @Override
     public void restart() throws IOException, InterruptedException {
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.cleanUp();
```

Comment:
What about try/catch in case this throws for some reason? I think we'd like to continue with the restart in that case.